rotkehlchen/exchanges/bitfinex.py

```diff
@@ -0,0 +1,936 @@
+            log.error(msg)
+            return None, msg
+
+        if currency_map_response.success is False:
+            result, msg = self._process_unsuccessful_response(
+                response=currency_map_response.response,
+                case='balances',
+            )
+            return result, msg
+
+        response = self._api_query('wallets')
+        if response.status_code != HTTPStatus.OK:
+            result, msg = self._process_unsuccessful_response(
+                response=response,
+                case='balances',
+            )
+            return result, msg
+        try:
+            response_list = rlk_jsonloads_list(response.text)
+        except JSONDecodeError as e:
+            msg = f'{self.name} returned invalid JSON response: {response.text}.'
+            log.error(msg)
+            raise RemoteError(msg) from e
+
+        # Wallet items indexes
+        currency_index = 1
+        balance_index = 2
+        asset_balance: DefaultDict[Asset, Balance] = defaultdict(Balance)
+        asset_usd_price: Dict[Asset, Price] = {}
+        for wallet in response_list:
+            if len(wallet) < API_WALLET_MIN_RESULT_LENGTH or wallet[balance_index] <= 0:
+                continue
+
+            try:
+                asset = asset_from_bitfinex(
+                    bitfinex_name=wallet[currency_index],
+                    currency_map=currency_map_response.currency_map,
+                )
+            except (UnknownAsset, UnsupportedAsset) as e:
+                asset_tag = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                self.msg_aggregator.add_warning(
+                    f'Found {asset_tag} {self.name} asset {e.asset_name} due to: {str(e)}. '
+                    f'Ignoring its balance query.',
+                )
+                continue
+
+            if asset not in asset_usd_price:
+                try:
+                    usd_price = Inquirer().find_usd_price(asset=asset)
+                except RemoteError as e:
+                    self.msg_aggregator.add_error(
+                        f'Error processing {self.name} balance result due to inability to '
+                        f'query USD price: {str(e)}. Skipping balance result.',
+                    )
+                    continue
+
+                asset_usd_price[asset] = usd_price
+
+            amount = FVal(wallet[balance_index])
+            asset_balance[asset] += Balance(
+                amount=amount,
+                usd_value=amount * asset_usd_price[asset],
+            )
+
+        return dict(asset_balance), ''
+
+    def query_online_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> List[AssetMovement]:
+        """Return the account deposits and withdrawals on Bitfinex.
+
+        Endpoint documentation:
+        https://docs.bitfinex.com/reference#rest-auth-movements
+
+        TODO: Check if 'sort' by ascending works.
```

Comment:
how can we check that?

Reply:
Thanks to Phil we know it does not work, I need to address it.